graphlearn_torch/csrc/cpu/random_sampler.cc

```diff
@@ -175,4 +281,51 @@ void CPURandomSampler::UniformSample(const int64_t* col_begin,
   }
 }
 
+void CPURandomSampler::WeightedRandomSample(const int64_t* col_begin,
```

Comment:
WeightedRandom->Weighted cause weighted is not random.

Reply:
The source of randomness is from sample. The previous cases called random are actually of uniform distribution, while in this case the sampling should follow the multinomial distribution generated by the weights. Just the matter of replacing probability measure from uniform to another